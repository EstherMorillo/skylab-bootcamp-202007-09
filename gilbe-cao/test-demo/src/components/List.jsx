import React from 'react';
import { useState } from 'react';
import { useEffect } from 'react';
import productStore from '../stores/product.store';
import { addToCart, loadProducts } from '../actions/action-creators';

export default function List() {
	const [productList, setProductList] = useState(productStore.getProducts());

	useEffect(() => {
		productStore.escuchaCambios(actualizarLista);
		if (productList.length === 0) loadProducts();

		return () => productStore.dejeDeEscucharCambios(actualizarLista);
	}, [productList.length]);

	function actualizarLista() {
		setProductList(productStore.getProducts());
	}

	return (
		<>
			<p>there are {productList.length} items in the list</p>
			<button onClick={() => addToCart()}>to pa'yaaa !</button>
		</>
	);
}
