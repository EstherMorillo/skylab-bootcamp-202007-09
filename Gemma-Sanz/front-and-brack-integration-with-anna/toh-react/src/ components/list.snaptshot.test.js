import renderer from 'react-test-renderer';
import React from 'react';
import List from './list';
import { BrowserRouter as Router } from 'react-router-dom';

describe('App snapshot', () => {
	const tree = renderer.create(
		<Router>
			<List />
		</Router>
	);

	it('should match', () => {
		expect(tree.toJSON()).toMatchSnapshot();
	});
});
