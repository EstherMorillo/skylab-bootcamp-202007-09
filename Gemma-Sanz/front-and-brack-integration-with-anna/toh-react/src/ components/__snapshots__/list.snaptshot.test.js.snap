// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App snapshot should match 1`] = `
<div>
  <h2>
    My heroes
  </h2>
  <ul
    className="hero-list"
  />
  <a
    aria-current={null}
    href="/hero/"
    onClick={[Function]}
  >
    Create Hero
  </a>
</div>
`;
